java/amazon-corretto/Portfile

```diff
@@ -0,0 +1,143 @@
+    distfiles
+    patchfiles
+    supported_archs       noarch
+    use_configure         no
+    build {}
+    destroot {
+        set docdir ${destroot}${prefix}/share/doc/${subport}
+        xinstall -d ${docdir}
+        system "echo $subport is a stub port > ${docdir}/README"
+    }
+}
+
+subport corretto-openjdk8 {
+    # https://github.com/corretto/corretto-8/releases
+    supported_archs  x86_64
+
+    version      8.322.06.1
+    revision     0
+
+    description  Amazon Corretto OpenJDK 8 (Long Term Support)
+    long_description {*}${long_description_corretto}
+
+    master_sites https://corretto.aws/downloads/resources/${version}/
+
+    distname     amazon-corretto-${version}-macosx-x64
+    checksums    rmd160  98759066ee6e41884e3acbf7f9e7debd6a6d54d0 \
+                 sha256  b86a899aac567b86866fe3cff0b80ee7e7a53f93d9fbe1ec3a2c992376ff1cc6 \
+                 size    118882960
+
+    worksrcdir   amazon-corretto-8.jdk
+}
+
+subport corretto-openjdk11 {
+    # https://github.com/corretto/corretto-11/releases
+    supported_archs  x86_64
+
+    version      11.0.14.10.1
+    revision     0
+
+    description  Amazon Corretto OpenJDK 11 (Long Term Support)
+    long_description {*}${long_description_corretto}
+
+    master_sites https://corretto.aws/downloads/resources/${version}/
+
+    distname     amazon-corretto-${version}-macosx-x64
+    checksums    rmd160  39004f9cabd34ba08af06c37be3c8896bde2176d \
+                 sha256  d4b8e365091ebcadd3c70709c26ab01d7a58aeb57978e0d68a7805f4cee6be71 \
+                 size    187369666
+
+    worksrcdir   amazon-corretto-11.jdk
+}
+
+subport corretto-openjdk17 {
+    # https://github.com/corretto/corretto-17/releases
+    supported_archs  x86_64 arm64
+
+    version      17.0.2.8.1
+    revision     0
+
+    description  Amazon Corretto OpenJDK 17 (Long Term Support)
+    long_description {*}${long_description_corretto}
+
+    master_sites https://corretto.aws/downloads/resources/${version}/
+
+    distname     amazon-corretto-${version}-macosx-x64
+    if {${configure.build_arch} eq "x86_64"} {
+        distname     amazon-corretto-${version}-macosx-x64
+        checksums    rmd160  8a622fe2c8dd5eba07330804d7b98c3b67b5ee78 \
+                     sha256  eec3acc2d1d303ef23c79ed45a8374002e34c14e34ba0fa931e98cbf6b963345 \
+                     size    187776090
+    } elseif {${configure.build_arch} eq "arm64"} {
+        distname     amazon-corretto-${version}-macosx-aarch64
+        checksums    rmd160  e5eb0341468193a223ec927abade6f151831da06 \
+                     sha256  b6d3c359bd31a3c1697fe386c927e139bd72572e72cdc12a08b5e1144fe35438 \
+                     size    185518346
+    }
+
+    worksrcdir   amazon-corretto-17.jdk
+}
+
+if {[string match corretto-openjdk* ${subport}]} {
+    homepage     https://aws.amazon.com/corretto/
+}
+
+livecheck.type  none
+
+use_configure    no
+build {}
+
+if {[string match corretto-openjdk* ${subport}]} {
```

Comment:
This `if` shouldn't be necessary after the split, right? Also for the other new portfiles.